pkg/cache/clusterqueue_test.go

```diff
@@ -223,58 +214,183 @@ func TestCohortCanFit(t *testing.T) {
 				},
 			},
 			wantFit: false,
-		},
-		"missing flavor": {
-			c: &Cohort{
-				Name: "C",
-				RequestableResources: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU:    5,
-						corev1.ResourceMemory: 5,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
+						"f2": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
-				},
-				Usage: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU:    5,
-						corev1.ResourceMemory: 5,
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    4,
+							corev1.ResourceMemory: 4,
+						},
+						"f2": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    4,
+							corev1.ResourceMemory: 4,
+						},
 					},
 				},
+				ResourceGroups: nil,
 			},
+		},
+		"missing flavor": {
 			request: FlavorResourceQuantities{
 				"f2": map[corev1.ResourceName]int64{
 					corev1.ResourceCPU:    1,
 					corev1.ResourceMemory: 1,
 				},
 			},
 			wantFit: false,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
+					},
+				},
+				ResourceGroups: nil,
+			},
 		},
 		"missing resource": {
-			c: &Cohort{
-				Name: "C",
-				RequestableResources: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 5,
+			request: FlavorResourceQuantities{
+				"f1": map[corev1.ResourceName]int64{
+					corev1.ResourceCPU:    1,
+					corev1.ResourceMemory: 1,
+				},
+			},
+			wantFit: false,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 5,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 3,
+						},
 					},
 				},
-				Usage: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 3,
+				ResourceGroups: nil,
+			},
+		},
+		"lendingLimit disabled can fit": {
```

Comment:
Wouldn't this case duplicate with another case, `can fit`?

Reply:
Yes, this case is actually duplicated. We add this to show how lendingLimit works when enabled or disabled.
I can remove this if this is not necessary.

Reply:
IIUC, the `lendingLimit` feature is disabled in both cases (`can fit` and `lendingLimit disabled can fit`), and both cases verify the completely same situation.

If my understanding is correct, having one of the cases (`can fit` and `lendingLimit disabled can fit`) would be enough.

Reply:
Done.